src/translator/cache.h

```diff
@@ -0,0 +1,205 @@
+#ifndef SRC_BERGAMOT_CACHE_H_
+#define SRC_BERGAMOT_CACHE_H_
+
+#include <list>
+#include <mutex>
+#include <unordered_map>
+#include <utility>
+
+#include "3rd_party/L4/inc/L4/LocalMemory/HashTableService.h"
+#include "common/hash.h"
+#include "processed_request_sentence.h"
+#include "translator/beam_search.h"
+
+namespace marian {
+namespace bergamot {
+
+class TranslationModel;
+
+struct CacheKey {
+  const TranslationModel *model;
+  const marian::Words &words;
+};
+
+struct HashCacheKey {
```

Comment:
Why is this a struct, when it clearly has no state? Every time you call it with a the full `CacheKey`. Some comments on the above function as well.